share/eds/blockstore.go

```diff
@@ -0,0 +1,177 @@
+package eds
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/shard"
+	lru "github.com/hashicorp/golang-lru"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+var _ blockstore.Blockstore = (*Blockstore)(nil)
+
+var (
+	maxCacheSize            = 100
+	ErrUnsupportedOperation = errors.New("unsupported operation")
+	ErrMultipleShardsFound  = errors.New("found more than one shard with the provided cid")
+)
+
+type accessorWithBlockstore struct {
+	sa *dagstore.ShardAccessor
+	bs dagstore.ReadBlockstore
+}
+
+// Blockstore implements the blockstore.Blockstore interface on an EDSStore.
+// The lru cache approach is heavily inspired by the open PR filecoin-project/dagstore/116.
+// The main differences to the implementation here are that we do not support multiple shards per key,
+// call GetSize directly on the underlying RO blockstore, and do not throw errors on Put/PutMany.
+// Also, we do not abstract away the blockstore operations.
+type Blockstore struct {
+	store *EDSStore
+
+	bsStripedLocks [256]sync.Mutex
```

Comment:
can you doc what these are guarding and how they're indexed?

Reply:
@distractedm1nd why 256 again? cache size is 100 ? shouldn't this be equivalent to cache size? If so, extract into var and use for both instantiating cache and fixing the size here.